### PR TITLE
change internal._Filter.all and any to accept a list

### PR DESCRIPTION
**Pull request by gabrys**
Make _Filter.all, _Filter.any and _AssociativeOperator require non-empty list of sub-filters.
But allow alpha to work in backward-compatible way by introducing an _EmptyAssociativeOperator that generates an empty string as the generated query.

## Summary by Sourcery

Revise the internal filter API to use list-based signatures for combining filters and update all usages and related documentation accordingly.

Enhancements:
- Change `_Filter.all` and `_Filter.any` to accept a list of filters instead of variadic parameters
- Refactor `__and__`, `__or__` and `__invert__` overloads to use the new list-based combinator signatures
- Update composite predicate methods (`matches_all`, `matches_none`, `contains_all`, `contains_none`) to build filters from lists
- Replace `name_matches_all`/`name_matches_none` parameters with a `must_match_any` list of `_AttributeNameFilter` in `_AttributeFilter` docs
- Adjust `resolve_runs_filter` internal logic to construct OR filters from a list

Tests:
- Update end-to-end retrieval tests to use list inputs for `_Filter.all` and `_Filter.any`

## Summary by Sourcery

Migrate filter combinators to a strictly list-based API and enforce non-empty inputs while preserving backward compatibility for zero-argument public filters in the alpha client.

Enhancements:
- Require `_Filter.all` and `_Filter.any` to accept a non-empty list of sub-filters and raise on empty or invalid inputs
- Introduce an `_EmptyAssociativeOperator` to support legacy zero-argument `Filter.all()`/`Filter.any()` in the alpha client by producing an empty query
- Refactor attribute filter logic and documentation to replace separate `name_matches_all`/`name_matches_none` parameters with a `must_match_any` list
- Update composite predicate and `name_in` methods to use the new list-based combinator signatures
- Remove variadic overloads for filter combination and consolidate `__and__`/`__or__`/`__invert__` to internal list-based logic

Tests:
- Refactor end-to-end and unit tests to supply list inputs to `all`/`any` combinators
- Add new alpha-specific tests to validate behavior of empty `Filter.all()` and `Filter.any()` usages raising errors or yielding empty queries

### CODE REVIEW

File: src/neptune_fetcher/alpha/filters.py [hunk 1]
```diff
@@ -427,6 +427,12 @@ def to_query(self) -> str:
         return f" {self.operator} ".join(filter_queries)
 
     def _to_internal(self) -> _filters._Filter:
+        if not self.filters:
+            # Importing here to avoid circular dependency issues
```
Comment (michalsosn) [inline src/neptune_fetcher/alpha/filters.py:431]: but an internal _AssociativeOperator is created just a line below and it just uses _filters without importing anything locally

File: src/neptune_fetcher/alpha/_internal.py [hunk 2]
```diff
@@ -133,3 +134,22 @@ def get_default_project_identifier(context: Optional[Context] = None) -> Project
     if not project:
         raise ValueError("No project is set in the context. Please set a project before calling this function.")
     return ProjectIdentifier(project)
+
+
+class _EmptyAssociativeOperator(_filters._AssociativeOperator):
```
[the remark this answers is not in the document]
Comment (gabrys) [reply src/neptune_fetcher/alpha/_internal.py:140]: This is the version of `_AssociativeOperator` specific for `alpha` where it's allowed to have it with empty filters, that's why it's called _EmptyAssociativeOperator. Internal `_AssociativeOperator` was improved to disallow the empty filter list.

Happy to take suggestions on this.

**Review by michalsosn — APPROVED** [review]
